### ISSUE TITLE
`tpf.to_lightcurve()` always returns an empty light curve for TESScut data

### ISSUE THREAD

**Issue by rebekah9969** [KeplerGO/lightkurve#791]
I have been playing around with 
"Performing Pixel Level Decorrelation with the PLDCorrector in lightkurve" tutorial in colab and have been applying it to TESS data. I have noticed several potential issues - I apologize in advance if these are not issue or bugs, but simple mistakes made on my end. 

I first tried to run the task via creating a cut out from FFI data and then passing  through to PLDcorrector, but this brought up several issues. 

The code I used is the following, 

```
target = 'KIC 8462852' 
tpf = lk.search_tesscut(target, sector=15).download(cutout_size=(10, 10))
pld = lk.PLDCorrector(tpf)
corrected_lc = pld.correct()
raw_lc = tpf.to_lightcurve()
ax = raw_lc.normalize().scatter(c='r', alpha=0.5, label='Raw Flux')
corrected_lc.normalize().remove_outliers().scatter(ax=ax, label='PLD-corrected Flux');
```
This brings up a warning 
```
Warning: aperture mask contains zero pixels.
```

if I run
```
%debug
```
I get 

```
     13         raise NotImplementedError("This is an abstract base class.")
     14 
     15     def diagnose(self):
     16         """Returns plots which elucidate the most recent call to `correct()`."""
---> 17         raise NotImplementedError("This is an abstract base class.")
```
The light curve produced does not look correct, with no raw flux.
Now if I switch to using a tpf from MAST instead so...

```
tpf = lk.search_targetpixelfile(target, sector=15).download()
```

and run through in the same way. I get no warning when creating the light curves, however when I try to run 

```
pld.diagnose();
```

I get the following output 

```
---------------------------------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
<ipython-input-8-48eda9cd8b1c> in <module>
----> 1 pld.diagnose();

~/Desktop/TESS_LK/lightkurve/docs/source/lightkurve/lightkurve/correctors/corrector.py in diagnose(self)
     15     def diagnose(self):
     16         """Returns plots which elucidate the most recent call to `correct()`."""
---> 17         raise NotImplementedError("This is an abstract base class.")

NotImplementedError: This is an abstract base class.
```

**Comment by rebekah9969** [comment]
I have installed the latest version of lightkurve and can now get pld.diagnose(); to run if I use tpf. The package still does not seem to work properly for TPFs created from FFIs.

```
target = 'KIC 8462852' 
tpf = lk.search_tesscut(target, sector=15).download(cutout_size=(10, 10))
pld = lk.PLDCorrector(tpf)
corrected_lc = pld.correct()
raw_lc = tpf.to_lightcurve()
ax = raw_lc.normalize().scatter(c='r', alpha=0.5, label='Raw Flux')
corrected_lc.normalize().remove_outliers().scatter(ax=ax, label='PLD-corrected Flux');
```

produces a warning 

```
Warning: aperture mask contains zero pixels.
/Users/hounsell/opt/anaconda3/lib/python3.7/site-packages/numpy/lib/nanfunctions.py:1115: RuntimeWarning: All-NaN slice encountered
  overwrite_input=overwrite_input)
<string>:6: RuntimeWarning: Degrees of freedom <= 0 for slice.
```
There still does not seem to be a raw light-curve


```
pld.diagnose();
```

Does seem to work, but the output does not seem to be correct.

**Comment by rebekah9969** [comment]
Apologies
target = 'KIC 8462852'

**Comment by barentsen** [comment]
It looks like the issue here is that `tpf.to_lightcurve()` always returns an empty light curve for TESScut data.

This is because the `tpf.extract_aperture_photometry()` helper function defaults to using `aperture_mask='pipeline'`, but there never is a pipeline mask for TESScut data.

The workaround is for the user to call `tpf.to_lightcurve(aperture_mask="threshold")`.  Of course that's not user-friendly!  I guess we should modify Lightkurve to automatically default to using the threshold mask if the pipeline mask is empty!  Thoughts? Volunteers?

**Comment by barentsen** [comment]
This issue has been fixed by #812. Closing!
